gossip/gossip.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p"
+)
+
+// GossipableAny exists to help create non-nil pointers to a concrete Gossipable
+type GossipableAny[T any] interface {
+	*T
+	Gossipable
+}
+
+type Config struct {
+	Frequency time.Duration
+	PollSize  int
+}
+
+func NewGossiper[T any, U GossipableAny[T]](
+	config Config,
+	set Set[U],
+	client *p2p.Client,
+	codec codec.Manager,
+	codecVersion uint16,
```

Comment:
I don't think we should be using the `codec` for this. We should be using proto for all p2p messages going forward.

Reply:
Ported stuff over to use proto